static/js/resources.js

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+/*global $ */
+
+$(document).ready(function () {
+    
+    'use strict';
+
+    /** Shows or hides the contents of a category.
+     * @param elm {String} The selector of an element whose contents will be toggled
+     * @param state {Boolean} True to toggle open, false to toggle closed
+     */
+    var toggleCategory = function (elm, state) {
+        var element = $(elm);
+        element
+            .attr('aria-expanded', state)
+            .closest('.resource-section')
+            .toggleClass('resources-content-show', state);
+        // Change toggle indicator for category being toggled
+        element.closest('section').find('div').toggleClass('collapsed', !state);
+    };
+
+    /** When "show all" button is clicked, expands all categories to show their contents. */
+    $('.resources-showAll').click(function () {
+        toggleCategory('.resources-toggleItem', true);
+    });
+
+    /** When "hide all" button is clicked, collapses all categories to hide their contents. */
+    $('.resources-hideAll').click(function () {
+        toggleCategory('.resources-toggleItem', false);
+    });
+
+    /** Toggle each category to hide or show their respective content(s).
+     * @param (evt) A jQuery event, happens each time the object is clicked
+     */
+    $('.resources-toggleItem').click(function (evt) {
+        var target = $(evt.delegateTarget);
+        var state = target.attr('aria-expanded') === 'true' ? true : false;
+        toggleCategory(target, !state);
+        // Remove focus indicator after clicks so that it doesn't linger indefinitely
+        this.blur();
```

Comment:
@michelled Is this a bad idea? Without this if you expand a resource then collapse it the focus indicator remains visible.